integrations/oci/tls-certificates/src/main/java/io/helidon/integrations/oci/tls/certificates/DefaultOciCertificatesDownloader.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.oci.tls.certificates;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+
+import io.helidon.integrations.oci.sdk.runtime.OciExtension;
+import io.helidon.integrations.oci.tls.certificates.spi.OciCertificatesDownloader;
+
+import com.oracle.bmc.certificates.CertificatesClient;
+import com.oracle.bmc.certificates.requests.GetCertificateAuthorityBundleRequest;
+import com.oracle.bmc.certificates.requests.GetCertificateBundleRequest;
+import com.oracle.bmc.certificates.responses.GetCertificateAuthorityBundleResponse;
+import com.oracle.bmc.certificates.responses.GetCertificateBundleResponse;
+import jakarta.inject.Singleton;
+
+import static io.helidon.integrations.oci.tls.certificates.spi.OciCertificatesDownloader.create;
+
+/**
+ * Implementation of the {@link OciCertificatesDownloader} that will use OCI's Certificates Service to download certs.
+ */
+@Singleton
+class DefaultOciCertificatesDownloader implements OciCertificatesDownloader {
+
+    @Override
+    public Certificates loadCertificates(String certOcid) {
+        Objects.requireNonNull(certOcid);
+        try {
+            return loadCerts(certOcid);
+        } catch (Exception e) {
```

Comment:
general catch clause should not be done - if there is a checked exception, catch it. Otherwise only wrap exceptions that you need to wrap.
If the method may throw our runtime exceptions, re-throw those without wrapping